sql/src/main/java/io/crate/planner/node/ddl/RefreshTablePlan.java

```diff
@@ -0,0 +1,139 @@
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.planner.DependencyCarrier;
+import io.crate.planner.Plan;
+import io.crate.planner.PlannerContext;
+import io.crate.planner.operators.SubQueryResults;
+import io.crate.sql.tree.Assignment;
+import io.crate.sql.tree.Table;
+import io.crate.types.DataTypes;
+import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static io.crate.data.SentinelRow.SENTINEL;
+
+public class RefreshTablePlan implements Plan {
+
+    private final RefreshTableAnalyzedStatement analysis;
+
+    public RefreshTablePlan(RefreshTableAnalyzedStatement analysis) {
+        this.analysis = analysis;
+    }
+
+    @Override
+    public void executeOrFail(DependencyCarrier dependencies,
+                              PlannerContext plannerContext,
+                              RowConsumer consumer,
+                              Row parameters,
+                              SubQueryResults subQueryResults) {
+        if (analysis.tables().isEmpty()) {
+            consumer.accept(InMemoryBatchIterator.empty(SENTINEL), null);
+            return;
+        }
+
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            plannerContext.transactionContext(),
+            plannerContext.functions(),
+            x,
+            parameters,
+            subQueryResults
+        );
+
+        ArrayList<String> toRefresh = new ArrayList<>();
+        for (Map.Entry<Table<Symbol>, DocTableInfo> table : analysis.tables().entrySet()) {
+            var tableInfo = table.getValue();
+            var tableSymbol = table.getKey();
+            if (!tableInfo.isPartitioned()) {
+                toRefresh.addAll(Arrays.asList(tableInfo.concreteIndices()));
+            } else {
+                var partitionName = toPartitionName(
+                    tableInfo,
+                    tableSymbol.partitionProperties(),
+                    eval);
+                if (!tableInfo.partitions().contains(partitionName)) {
+                    throw new PartitionUnknownException(partitionName);
+                }
+                toRefresh.add(partitionName.asIndexName());
+            }
+        }
+
+        RefreshRequest request = new RefreshRequest(toRefresh.toArray(String[]::new));
+        request.indicesOptions(IndicesOptions.lenientExpandOpen());
+
+        var transportRefreshAction = dependencies.transportActionProvider().transportRefreshAction();
+        transportRefreshAction.execute(
+            request,
+            new OneRowActionListener<>(
+                consumer,
+                response -> new Row1((long) analysis.tables().size())
+            )
+        );
+    }
+
+    private static PartitionName toPartitionName(DocTableInfo tableInfo,
```

Comment:
This logic is similar to `PartitionPropertiesAnalyzer#toPartitionName`,  i'll extract it later once it will be required for other plans.

Reply:
This looks more like `AlterTableAnalyzer.createPartitionName`. This method is already used in other analyzers, I've created already a variant usable with `List<Assignment<Symbol>>` while implementing unbound alter table (not pushed yet). We could replace that later on as well..

Reply:
Pushed it now, see https://github.com/crate/crate/pull/9195/files#diff-af900ac13503950089e585847aeda467R179

Reply:
Thank! 👍 I was just wondering whether we might want to create a separate class for holding these helper methods somewhere in the planner package and slowly drop the whole `PartitionPropertiesAnalyzer` as we implement unbound analysis for other statements, to have less deps on analyzer in planner or we do it later once we move a bit further with unbound analyzers?